kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java

```diff
@@ -0,0 +1,133 @@
+			isSuccessful &= this.getLaborLedgerPendingEntries().addAll(expensePendingEntries);
+		}
+
+		List<LaborLedgerPendingEntry> benefitPendingEntries = LaborPendingEntryGenerator.generateBenefitPendingEntries(this, expenseTransferAccountingLine, sequenceHelper);
+		if (benefitPendingEntries != null && !benefitPendingEntries.isEmpty()) {
+			isSuccessful &= this.getLaborLedgerPendingEntries().addAll(benefitPendingEntries);
+		}
+
+		String salaryObjectCode = getParameterService().getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborParameterKeyConstants.SalaryExpenseTransfer.TUITION_REMISSION_DIRECT_BILL_SALARY_PARM_NM);
+		String fringeBenefitObjectCode = getParameterService().getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborParameterKeyConstants.SalaryExpenseTransfer.TUITION_REMISSION_DIRECT_BILL_POSTING_PARM_NM);
+
+		if (StringUtils.isNotEmpty(salaryObjectCode) && StringUtils.isNotEmpty(fringeBenefitObjectCode) && StringUtils.equals(expenseTransferAccountingLine.getFinancialObjectCode(), salaryObjectCode)) {
+			KualiDecimal benefitAmount = calculateTuitionRemission(expenseTransferAccountingLine, salaryObjectCode, fringeBenefitObjectCode);
+
+			if (benefitAmount.isNonZero()) {
+				benefitPendingEntries = LaborPendingEntryGenerator.generateBenefitPendingEntries(this, expenseTransferAccountingLine, sequenceHelper, benefitAmount, fringeBenefitObjectCode);
+				if (benefitPendingEntries != null && !benefitPendingEntries.isEmpty()) {
+					isSuccessful &= this.getLaborLedgerPendingEntries().addAll(benefitPendingEntries);
+				}
+			}
+		}
+		
+		return isSuccessful;
+	}
+
+	public KualiDecimal calculateTuitionRemission(ExpenseTransferAccountingLine expenseTransferAccountingLine, String salaryObjectCode, String fringeBenefitObjectCode) {
+		BigDecimal salaryAmount = amountForCalculation(expenseTransferAccountingLine, salaryObjectCode);
+		BigDecimal tuitionRemissionAmount = amountForCalculation(expenseTransferAccountingLine, fringeBenefitObjectCode);
+
+		if (salaryAmount == null || salaryAmount.intValue() == 0 || tuitionRemissionAmount == null || tuitionRemissionAmount.intValue() == 0) {
+			return KualiDecimal.ZERO;
+		}
+
+		BigDecimal fringeBenefitPercent = tuitionRemissionAmount.divide(salaryAmount, 12, BigDecimal.ROUND_HALF_UP);
+		return new KualiDecimal(fringeBenefitPercent.multiply(expenseTransferAccountingLine.getAmount().bigDecimalValue()));
+	}
+
+	public BigDecimal amountForCalculation(ExpenseTransferAccountingLine expenseTransferAccountingLine, String objectCode) {
+		Map<String, String> fieldValues = new HashMap<String, String>();
+		fieldValues.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, expenseTransferAccountingLine.getPayrollEndDateFiscalYear().toString());
+		fieldValues.put(KFSPropertyConstants.EMPLID, expenseTransferAccountingLine.getEmplid());
+		fieldValues.put(KFSPropertyConstants.FINANCIAL_OBJECT_CODE, objectCode);
```

Comment:
Yikes, this lookup needs to include position number (see KATTS-3656). You've no idea what trouble this caused when this mod originally went live (it was a UA president requested mod).